lib/purchases_flutter.dart

```diff
@@ -175,13 +175,13 @@ class Purchases {
   /// [upgradeInfo] Android only. Optional UpgradeInfo you wish to upgrade from
   /// containing the oldSKU and the optional prorationMode.
   static Future<PurchaserInfo> purchasePackage(Package packageToPurchase,
-      {UpgradeInfo upgradeInfo}) async {
+      {UpgradeInfo? upgradeInfo}) async {
     var response = await _channel.invokeMethod('purchasePackage', {
       'packageIdentifier': packageToPurchase.identifier,
       'offeringIdentifier': packageToPurchase.offeringIdentifier,
       'oldSKU': upgradeInfo != null ? upgradeInfo.oldSKU : null,
       'prorationMode': upgradeInfo != null && upgradeInfo.prorationMode != null
```

Comment:
do we need to be doing `upgradeInfo!.prorationMode` here?

Reply:
It's not needed here as `upgradeInfo` is a function parameter and thus the analyzer can statically make sure that it's never `null`.

Unwrapping would cause a compilation error.

Reply:
More information about this one:
https://github.com/dart-lang/sdk/issues/42626

Reply:
that makes sense 👍